### PR TITLE
[wallet] Refactor to use new MIN_CHANGE

### PR DESCRIPTION
**Pull request by maflcko**
Introduce a new constant `MIN_CHANGE` and use it in appropriate places.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
I will add some more tests later.

**Comment by MarcoFalke** [comment]
ping @dooglus: As you are one of the authors of the coin selection code and tests, would you mind giving a concept ACK or concept NACK on this pull?

**Comment by dooglus** [comment]
Looks good to me. It makes sense to have the minimum change not be hardcoded to 0.01.

**Comment by btcdrak** [comment]
utACK

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -41,8 +41,12 @@ extern bool fPayAtLeastCustomFee;
 static const CAmount DEFAULT_TRANSACTION_FEE = 0;
 //! -paytxfee will warn if called with a higher fee than this amount (in satoshis) per KB
 static const CAmount nHighTransactionFeeWarning = 0.01 * COIN;
+//! -mintxfee default
+static const CAmount DEFAULT_TRANSACTION_MINFEE = 1000;
 //! -maxtxfee default
 static const CAmount DEFAULT_TRANSACTION_MAXFEE = 0.1 * COIN;
+//! minimum change amount
+static const CAmount nMinChange = CENT;
```
Comment (sipa) [inline src/wallet/wallet.h]: If it's a constant, please use an all-upper-case name to make that clear, like MINIMUM_CHANGE.

If you plan to turn it into a runtime variable later, it's probably better to make it a runtime variable immediately now, and initialize it with a constant, for example DEFAULT_MINIMUM_CHANGE.

Comment (MarcoFalke) [inline src/wallet/wallet.h]: No plans yet to turn it into a runtime variable.

**Comment by MarcoFalke** [comment]
Addressed nits by @sipa in follow up commit.

**Comment by wtogami** [comment]
Add mention of DEFAULT_TRANSACTION_MINFEE to the commit message?

**Comment by MarcoFalke** [comment]
@wtogami I think you intended to put the comment over at #6696?

**Comment by MarcoFalke** [comment]
Anything holding this back? Does this need more review or additional test cases?

**Comment by wtogami** [comment]
@MarcoFalke oh?  it appeared that the commits in this PR added it.

**Comment by sipa** [comment]
Concept & light code review ACK.

**Comment by MarcoFalke** [comment]
@wtogami Done. (Fixed commit msg)

**Comment by laanwj** [comment]
Concept ACK. After this we could completely get rid of CENT.

File: src/wallet/wallet.h [hunk 2]
```diff
@@ -41,8 +41,12 @@ extern bool fPayAtLeastCustomFee;
 static const CAmount DEFAULT_TRANSACTION_FEE = 0;
 //! -paytxfee will warn if called with a higher fee than this amount (in satoshis) per KB
 static const CAmount nHighTransactionFeeWarning = 0.01 * COIN;
+//! -mintxfee default
+static const CAmount DEFAULT_TRANSACTION_MINFEE = 1000;
```
Comment (laanwj) [inline src/wallet/wallet.h]: Should use this new default constant in init.cpp as well (e.g. HelpMessage)

**Comment by dcousens** [comment]
utACK

File: src/wallet/wallet.h (hunk 2 above)
Comment (MarcoFalke) [inline src/wallet/wallet.h]: @laanwj Done

**Pull request merged**